internal/database/dbconn/open.go

```diff
@@ -69,27 +73,144 @@ func openDBWithStartupWait(cfg *pgx.ConnConfig) (db *sql.DB, err error) {
 	}
 }
 
+// extendedDriver wraps sqlHooks' driver to provide a conn that implements Ping, ResetSession
+// and CheckNamedValue, which is mandatory as otelsql is instrumenting these methods.
```

Comment:
```suggestion
// and CheckNamedValue, which is mandatory as otelsql is instrumenting these methods.
// For all mandatory methods the sqlHooks driver is used. For the optional methods namely Ping, ResetSession and CheckNamedValue 
// (which the sqlHooks driver does not implement), extendedConn goes to the original default driver.
```
This is more for myself to make sure I understand the diagram here @jhchabran